web/packages/teleport/src/Discover/Shared/SetupAccess/useUserTraits.ts

```diff
@@ -98,6 +104,19 @@ export function useUserTraits() {
       }
       break;
 
+    // Note: specific to AWS CLI access
+    case ResourceKind.Application:
+      if (resourceSpec.appMeta.awsConsole) {
+        const { awsRoles } = (agentMeta as AppMeta).app;
+        staticTraits.awsRoleArns = arrayStrDiff(
+          awsRoles.map(r => r.arn),
+          dynamicTraits.awsRoleArns
+        );
+        break;
+      }
+
+    // allow fall through
+    // eslint-disable-next-line no-fallthrough
```

Comment:
I'm not sure if this is a good idea to throw the generic error, its message is misleading here - we actually support `ResourceKind.Application`, but not that particular type.
Maybe it would be better to throw an error explicitly? Btw, I wouldn't add `useUserTraits.ts:` prefix, it's not something the user should see.